### PR TITLE
[New Package] samurai

### PR DESCRIPTION
**Pull request by saltedcoffii**
Samurai is basically ninja rewritten in C. I'll probably go through and convert all the package files to samurai since it does build faster than ninja does.

### CODE REVIEW

**Comment by satmandu** [comment]
I would hesitate on converting all packages. Speed is nice, but if software is packaged to be built with certain tools, it helps when debugging to be building with those tools... If you can get package upstream to support samurai... then that's another issue.

File: packages/samurai.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  homepage 'https://github.com/michaelforney/samurai'
+  version '1.2'
+  license 'Apache-2.0 and MIT'
+  compatibility 'all'
+  source_url 'https://github.com/michaelforney/samurai/releases/download/1.2/samurai-1.2.tar.gz'
+  source_sha256 '3b8cf51548dfc49b7efe035e191ff5e1963ebc4fe8f6064a5eefc5343eaf78a5'
+
+  def self.patch
+    system "sed -i 's:PREFIX=/usr/local:PREFIX=#{CREW_PREFIX}:' Makefile"
+    system "sed -i 's:MANDIR=\$(PREFIX)/share/man:MANDIR=#{CREW_MAN_PREFIX}:g' Makefile"
```
Comment (uberhacker) [inline packages/samurai.rb:14]: Could we set these in the build section instead?  Something like:
```
system "PREFIX=#{CREW_PREFIX} MANDIR=#{CREW_MAN_PREFIX} \
            make CFLAGS='-pipe -O2 -flto=auto -fuse-ld=gold' \
            LDFLAGS='-flto=auto -fuse-ld=gold' \
            CC='#{CREW_TGT}-gcc'"
```

Comment (saltedcoffii) [reply packages/samurai.rb:14]: Not with the way the makefile is written, no. I already tried that.

**Comment by saltedcoffii** [comment]
Oh sorry, I meant samurai is a drop in replacement for ninja. Some distros, like Alpine Linux, use samurai in replacement of ninja entirely. They use the same syntax and everything, it's like using llvm-strip instead of gnu strip.

File: packages/samurai.rb (hunk 1 above)
Comment (uberhacker) [reply packages/samurai.rb:14]: I would be surprised if it didn't work.

Comment (saltedcoffii) [reply packages/samurai.rb:14]: Becuase it's `PREFIX=/usr/local` not `PREFIX:=/usr/local` it doesn't work. Same with MANDIR

Comment (uberhacker) [reply packages/samurai.rb:14]: Ah, that makes sense now.

**Pull request merged**